### DISCUSSION TITLE
Add a "new" way for duplicating scripts to other sprites

### DISCUSSION THREAD

**Discussion by sfederici** [ScratchAddons/ScratchAddons#7427] [Ideas]
My students keep making "mistakes" when duplicating scripts since we moved to Scratch 3 (we were using Scratch 1.4 until ScratchAddons was out, as I never liked Scratch 2 for teaching/learning).

In Scratch 1.4, to duplicate a script in another sprite, you have two ways:
1. drag the script to another sprite thumbnail in the sprite corral
2. select "duplicate" in the right-click menu, drag the script (that is attached to the mouse pointer) to another sprite thumbnail in the sprite corral, and click the thumbnail

While in Scratch 3 the first method works, the second doesn't. Indeed, after duplicating the script via right-click menu, dragging it to a different sprite thumbail in the sprite corral and clicking it, what you get is the sprite thumbnail being grasped and attached to the mouse pointer.

As the second method, even if redundant, is very natural (it is what my students do before I explain to them that they don't have to use the "duplicate" option in the right-click menu) my proposal is adding it as a secondary way of duplicating scripts to other sprites.

**Comment by WorldLanguages** [comment]
Very interesting. The current behavior isn't really intuitive.
